### PR TITLE
chore: adding doc to lib

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

Adding some documentation about the `topos-tce-broadcast` crate

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: scripts/check_readme.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+set -e
+
+param=$1
+
+function check {
+  if [ "$param" == "generate" ]; then
+    cargo readme -r $1 > $1/README.md
+  else
+    diff <(cargo readme  -r $1) $1/README.md || (echo 1>&2 "Please update the $1/README with "'`'"cargo readme -r $1 > $1/README.md"'`' && exit 1 )
```
[the remark this answers is not in the document]
Comment (Freyskeyd) [reply scripts/check_readme.sh:11]: That's the case

**Review by dvdplm — APPROVED** [review]
Yes please!

**Review by hadjiszs — APPROVED** [review]
also several occurrences of `broadcasted` where `broadcast` seems better.

File: crates/topos-tce-broadcast/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+The Broadcast allows a set of validators to agree on a set of messages in order to
+reach consensus about the delivery of a certificate.
+
+Each certificates need to be broadcasted to the network, and each validator needs to
+receive a set of messages from the other validators in order to reach a threshold.
+The threshold is defined by the `ReliableBroadcastParams` and is the minimum number of
+validators that need to agree on a certificate in order to consider it delivered.
+
+This crate is responsible for validating and driving the broadcast of every certificates.
```
[the remark this answers is not in the document]
Comment (Freyskeyd) [reply crates/topos-tce-broadcast/README.md:19]: I think we can add this kind of doc later to improve and explain how it settles the `authenticated` message

**Pull request merged**